oggsound/src/OgreOggListener.cpp

```diff
@@ -36,23 +36,29 @@
 
 namespace OgreOggSound
 {
-#if OGGSOUND_THREADED
+#if OGRE_THREAD_PROVIDER
+/*
 #	ifdef POCO_THREAD
 	Poco::Mutex OgreOggSound::OgreOggListener::mMutex;
 #	else
 	boost::recursive_mutex OgreOggSound::OgreOggListener::mMutex;
 #	endif
+*/
+	OGRE_STATIC_MUTEX_INSTANCE(OgreOggSound::OgreOggListener::mMutex);
 #endif
 
 	/*/////////////////////////////////////////////////////////////////*/
 	void OgreOggListener::setPosition(ALfloat x, ALfloat y, ALfloat z)
 	{
-#if OGGSOUND_THREADED
+#if OGRE_THREAD_PROVIDER
```

Comment:
for merging you should delete any commented code and redundant #if

here, OGRE_LOCK_MUTEX_NAMED will be a noop if OGRE_THREAD_PROVIDER == 0